pkg/monitor/cluster/nsgreconciliations.go

```diff
@@ -0,0 +1,53 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	subnetscontroller "github.com/Azure/ARO-RP/pkg/operator/controllers/subnets"
+)
+
+func (mon *Monitor) emitNSGReconciliation(ctx context.Context) error {
+	co, err := mon.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	reconciled := 0
+	updated, err := annotationUpdated(co.Annotations)
+	if err != nil {
+		return err
+	}
+
+	if updated {
+		reconciled = 1
+	}
+
+	mon.emitGauge("nsg.reconciliations", int64(reconciled), nil)
+
+	return nil
```

Comment:
this could be simplified I think 
```suggestion
	updated, err := annotationUpdated(co.Annotations)
	if err != nil {
		return err
	}

	if updated {
            mon.emitGauge("nsg.reconciliations", int64(1), nil)
	}


	return nil
```

`emitGauge(...,0)` is equivalent to not calling it I think, since it is kind of a counter

Reply:
There are different sampling strategies provided by Jarvis/Geneva, such as Sum, Count, and Avg. If we use only the Sum sampling method, we can neglect zeros emitted by this metric, but we will not be able to use other sampling methods. I am not sure if we actually need the other methods, but I am also not sure if it is good practice to simplify the code at the expense of sacrificing the ability to use different sampling methods.

@gvanderpotte, what are your thoughts?

Reply:
I think that in that precise case, we can put "fill with 0s" when missing values, so i think it's equivalent in this precise case. As JEremy said, maybe let's not overflow Geneva if the value is normal (that's what we do for other metrics, we send only the value if it's abnormal state)